shoop/campaigns/admin_module/form_sets.py

```diff
@@ -58,11 +58,24 @@ def get_queryset(self):
         return self.owner.conditions.instance_of(self._get_actual_model())
 
 
-class BasketEffectsFormSet(BaseFormset):
+class BasketDiscountEffectsFormSet(BaseFormset):
     model = BasketDiscountEffect
 
     def get_queryset(self):
-        return self.owner.effects.instance_of(self._get_actual_model())
+        return self.owner.basket_discount_effects.instance_of(self._get_actual_model())
```

Comment:
This still needs to be instance_of sice we want the queryset for the subclass. The ``BasketDiscountEffect`` is still a base class for ``BasketDiscountAmount`` and ``BasketDiscountPercentage``.